backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -101,6 +103,26 @@ public FileDownloadResponse downloadFile(UUID fileId) {
         fileMetadataEntity.getName(), fileMetadataEntity.getType(), fileContents);
   }
 
+  @Override
+  public List<FileResponse> getAllFilesFromRoot(UUID bucketId) {
+    List<FileMetadataEntity> allFiles =
+        fileMetadataRepository.findAllByBucketIdAndParentFolderIsNull(bucketId);
+    return getFileResponses(allFiles);
+  }
+
+  @Override
+  public List<FileResponse> getAllFilesFromParent(UUID parentFolderId) {
+    List<FileMetadataEntity> allFiles =
+        fileMetadataRepository.findAllByParentFolderId(parentFolderId);
+    return getFileResponses(allFiles);
+  }
+
+  private List<FileResponse> getFileResponses(List<FileMetadataEntity> allFiles) {
```

Comment:
this one, maybe getFiles is better, right? `getFileReponses`  me personally leads me to think that there will be some response from files, I know it sounds not logicall when you put that in a sentence, but you get my point :)

Reply:
I get it. Will change it. :)